plugins/modules/ndo_mac_sec_policy.py

```diff
@@ -0,0 +1,424 @@
+
+            if security_policy and match.details.get("macsecParams")["securityPol"] != security_policy:
+                ops.append(
+                    dict(op="replace", path="{0}/{1}/macsecParams/securityPol".format(path, match.index), value=NDO_SECURITY_POLICY_MAP.get(security_policy))
+                )
+                match.details["macsecParams"]["securityPol"] = NDO_SECURITY_POLICY_MAP.get(security_policy)
+
+            if sak_expiry_time and match.details.get("macsecParams")["sakExpiryTime"] != sak_expiry_time:
+                ops.append(dict(op="replace", path="{0}/{1}/macsecParams/sakExpiryTime".format(path, match.index), value=sak_expiry_time))
+                match.details["macsecParams"]["sakExpiryTime"] = sak_expiry_time
+
+            if interface_type == "access":
+                if confidentiality_offset and match.details.get("macsecParams")["confOffSet"] != confidentiality_offset:
+                    ops.append(
+                        dict(op="replace", path="{0}/{1}/macsecParams/confOffSet".format(path, match.index), value="offset" + str(confidentiality_offset))
+                    )
+                    match.details["macsecParams"]["confOffSet"] = "offset" + str(confidentiality_offset)
+
+                if key_server_priority and match.details.get("macsecParams")["keyServerPrio"] != key_server_priority:
+                    ops.append(dict(op="replace", path="{0}/{1}/macsecParams/keyServerPrio".format(path, match.index), value=key_server_priority))
+                    match.details["macsecParams"]["keyServerPrio"] = key_server_priority
+
+            if mac_sec_keys:
+                mac_sec_keys_list = []
+                for mac_sec_key in mac_sec_keys:
+                    mac_sec_keys_list.append(
+                        dict(
+                            keyname=mac_sec_key.get("key_name"),
+                            psk=mac_sec_key.get("psk"),
+                            start=mso.verify_time_format(mac_sec_key.get("start_time")) if mac_sec_key.get("start_time") else None,
+                            end=mso.verify_time_format(mac_sec_key.get("end_time")) if mac_sec_key.get("end_time") else None,
+                        )
+                    )
+
+                ops.append(dict(op="replace", path="{0}/{1}/macsecKeys".format(path, match.index), value=mac_sec_keys_list))
+                match.details["macsecKeys"] = mac_sec_keys
+            elif mac_sec_keys == []:
+                # remove mac_sec_keys if the list is empty
+                ops.append(dict(op="remove", path="{0}/{1}/macsecKeys".format(path, match.index)))
+                match.details.pop("macsecKeys")
+
+            mso.sanitize(match.details)
+
+        else:
+            mac_sec_param_map = {}
+
+            payload = {"name": mac_sec_policy, "templateId": mso_template.template.get("templateId"), "schemaId": mso_template.template.get("schemaId")}
+            payload["adminState"] = admin_state
```

Comment:
Is there a reason why admin_state is included in the payload all the time even when it's not required?

Reply:
nope fixed this